tools/scripts/run_code_generation.py

```diff
@@ -14,6 +14,7 @@
 import zipfile
 from concurrent.futures import ProcessPoolExecutor, wait, FIRST_COMPLETED, ALL_COMPLETED
 from pathlib import Path
+from typing import Optional
```

Comment:
this file has been updated in main, I'm afraid you need to rebase.

Reply:
I will do it after this merge, I need to rebase over 20 versions.